### PR TITLE
Expose stream parameter in public strings combine APIs

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Add stream parameter to public APIs:

- `cudf::strings::concatenate()` (x2)
- `cudf::strings::join_strings()`
- `cudf::strings::join_list_elements()` (x2)
- `cudf::strings::repeat_string()`
- `cudf::strings::repeat_strings()` (x2)

Also added stream gtests and fixed up some doxygen comments.

Reference #13744 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by ttnghia — APPROVED** [review]

File: cpp/include/cudf/strings/combine.hpp [hunk 1]
```diff
@@ -127,25 +129,25 @@ std::unique_ptr<column> join_strings(
...
- * @param strings_columns List of strings columns to concatenate.
+ * @param strings_columns List of strings columns to concatenate
  * @param separators Strings column that provides the separator for a given row
- * @param separator_narep String that should be used in place of a null separator for a given
- *        row. Default of invalid-scalar means no row separator value replacements.
- *        Default is an invalid string.
- * @param col_narep String that should be used in place of any null strings
- *        found in any column. Default of invalid-scalar means no null column value replacements.
- *        Default is an invalid string.
+ * @param separator_narep String to replace a null separator for a given row.
```
Comment (hyperbolic2346) [inline cpp/include/cudf/strings/combine.hpp:134]: There are inconsistencies for periods in param comments all through here. In some places, you are removing them, but in most places, you are using them. Should we be consistent with these?

**Review by hyperbolic2346 — APPROVED** [review]
Tiny nit

File: cpp/include/cudf/strings/combine.hpp (hunk 1 above)
Comment (davidwendt) [reply cpp/include/cudf/strings/combine.hpp:134]: This has come up many times before. We should use periods only for sentences. 
I will not be able to make everyone happy so I'm going with Mark's recommendation here: https://github.com/rapidsai/cudf/pull/14247#discussion_r1349788189

Comment (ttnghia) [reply cpp/include/cudf/strings/combine.hpp:134]: If I am to be an owner of some new repository, I will write an official dev guide from the beginning: "All @params must be followed by complete sentence(s) that begin with a capitalized letter and end with period" :smile:

Comment (hyperbolic2346) [reply cpp/include/cudf/strings/combine.hpp:134]: Fair enough. We have thought about it and are following a plan. I am fine with this.

**Pull request merged**